software/control/serial_peripherals.py

```diff
@@ -349,13 +364,23 @@ def set_illumination_iris(self, value):
         self.serial_connection.write_and_check("J" + value + "\r", "J" + value, read_delay=3)
         return self.illumination_iris
 
+    def get_illumination_iris(self):
+        current_pos = self.serial_connection.write_and_check("rJ\r", "rJ", read_delay=0.01)
+        self.illumination_iris = int(int(current_pos[2:]) / 10)
```

Comment:
We're loosing resolution here. Probably ok for now but should make a note to change in the future.